book/data_management.md

```diff
@@ -3,56 +3,1853 @@
+
+commit 95b9016840e1d35bef0edf3afa06e41fdaaefce4
+Author: Russell Poldrack <poldrack@gmail.com>
+Date:   Sun Nov 16 12:12:08 2025 -0800
+
+    [DATALAD] new dataset
+```
+
+Here we see the commit messages that were automatically created by Datalad, first for creating the new dataset and then for downloading the URLS.  The `datalad download-url` function adds the URL to the log, which is useful for provenance tracking.
+
+#### Modifying files
+
+Now let's say that we want to make a change to one of the files and save the changes to the dataset.  Files tracked by Datalad are read-only ("locked") by default.  If we want to edit them, then we need to use `datalad unlock` to unlock the file:
+
+```bash
+➤  datalad unlock data/demographics.csv
+
+unlock(ok): data/demographics.csv (file)
+```
+
+We then use a Python script to make the change, which in this case is removing some columns from the dataset:
+
+```bash
+➤  python ../src/BetterCodeBetterScience/modify_data.py data/demographics.csv
+
+```
+
+We can now use `datalad status` to see that the file has been modified:
+
+```bash
+➤  datalad status
+ modified: data/demographics.csv (file)
+```
+
+And we can then save it using `datalad save`:
+```bash
+➤  datalad save -m "removed Motivation variables from demographics.csv"
+
+add(ok): data/demographics.csv (file)
+save(ok): . (dataset)
+action summary:
+  add (ok: 1)
+  save (ok: 1)
+```
+
+Datalad doesn't have a staging area like `git` does, so there is no need to first add and then commit the file; `datalad save` is equivalent to adding and then committing the changes. If we then check the status we see that there are no changes waiting to be saved:
+
+```bash
+➤  datalad status
+
+nothing to save, working tree clean
+```
+
+#### Pushing data to a remote repository
+
+Datalad is a particularly powerful tool for sharing data across systems.  It allows one to push or pull data from a number of different remote storage systems; in this example we will use the [Open Science Framework (OSF)](https://osf.io/) as our storage location, because it is particularly easy to use with Datalad.
+
+We first need to install and set up the `datalad-osf` Python package, per [the Datalad documentation](https://docs.datalad.org/projects/osf/en/latest/settingup.html).  We also need to create an account on the OSF site, and obtain a Personal Access Token for login.  We can then use Datalad to authenticate with OSF:
+
+```bash
+➤  datalad osf-credentials                                                1 ↵
+You need to authenticate with 'https://osf.io' credentials. https://osf.io/settings/tokens provides information on how to gain access
+token:
+osf_credentials(ok): [authenticated as Russell Poldrack <poldrack@stanford.edu>]
+```
+
+Having authenticated with OSF, we can now create a new OSF project using Datalad:
+
+```bash
+➤  datalad create-sibling-osf --title datalad-test-project -s osf
+create-sibling-osf(ok): https://osf.io/htprk/
```

Comment:
https://osf.io/htprk/ seems to be not reachable...  still private? I wondered if some public one would be published associated with this for demo purposes?  I would also be interested to see if we manage to auto-discover it in 
https://github.com/datalad/datalad-usage-dashboard/?tab=readme-ov-file#osf